src/resolvers/acl.js

```diff
@@ -37,5 +39,36 @@ export default {
 
       return dataSources.aclSourceDelete(args, context, parseResolveInfo(info))
     }
+  },
+
+  Acl: {
+    groupPermissions: async (source) => {
+      const { groupPermissions } = source
+
+      const camelcasedData = camelcaseKeys(groupPermissions, { deep: true })
+
+      return {
+        count: groupPermissions.length,
+        items: camelcasedData
+      }
+    }
+  },
+  GroupPermissions: {
+    group: async (source, args, context, info) => {
+      const { dataSources } = context
+      const resolvedInfo = parseResolveInfo(info)
+      const requestInfo = parseRequestedFields(resolvedInfo, {}, 'group')
+      const { groupId: id } = source
+
+      if (!id) {
+        return null
+      }
+
+      return dataSources.groupSourceFetch(
```

Comment:
I would add a check to requestedFields, if requested fields is only `id`, then you can `return { id }` here. We only need to execute the search if they have requested something we don't already have, and we have the id.